packages/lodestar/src/network/reqResp.ts

```diff
@@ -277,14 +279,23 @@ export class ReqResp extends (EventEmitter as IReqEventEmitterClass) implements
     const {libp2p, config, logger} = this;
     return (async function * () {
       const protocol = createRpcProtocol(method, encoding);
-      const {stream} = await libp2p.dialProtocol(peerId, protocol) as {stream: Stream};
+      let connectionTimer;
+      const {stream} = await Promise.race([
+        libp2p.dialProtocol(peerId, protocol),
```

Comment:
dial protocol accept third param options with abort signal :)

https://github.com/libp2p/js-libp2p/blob/v0.28.3/src/index.js#L316

Reply:
I would also extract to function that accepts peerId, protocol and timeout